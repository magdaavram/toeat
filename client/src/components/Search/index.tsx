import React from 'react';
import styled from 'styled-components';
import Button from 'components/Button';
import Input from 'components/Input';

interface IProps {
  handleSearch: (term: string) => void;
}

const Form = styled.form`
  display: flex;
  width: 70%;
  max-width: 1000px;
  height: 54px;
`;

const SearchInput = styled(Input)`
  height: 100%;
  border-width: 3px;
  border-radius: 12px;
  padding: 0 54px 0 18px;
`;

const SearchButton = styled(Button)`
  width: 30px;
  height: 30px;
  margin: auto 0;
  margin-left: -50px;
  opacity: 0.6;
`;

const Search = (props: IProps) => {
  const handleSearch = (e: any) => {
    const [input] = e.target.children;

    props.handleSearch(input.value);
    input.value = '';

    e.preventDefault();
  };

  return (
    <Form onSubmit={handleSearch}>
      <SearchInput type="search" placeholder={'Find recipes by ingredients or name...'} />
      <SearchButton type="submit" hasIcon={true} icon={'search'} iconWidth={27} iconHeight={27} />
    </Form>
  );
};

export default Search;
