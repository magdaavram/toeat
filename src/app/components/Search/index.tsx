import React from 'react';
import styled from 'styled-components';
import SearchIcon from '../../assets/icons/Search';
import Button from '../Button';
import Input from '../Input';

const Container = styled.div`
  display: flex;
  width: 70%;
  max-width: 1000px;
  height: 54px;
`;

const SearchInput = styled(Input)`
  height: 100%;
  border-width: 3px;
  border-radius: 12px;
  padding: 0 54px 0 18px;
`;

const SearchButton = styled(Button)`
  width: 30px;
  height: 30px;
  margin: auto 0;
  margin-left: -45px;
  opacity: 0.6;
`;

const Search = () => (
  <Container>
    <SearchInput placeholder={'Find recipes by ingredients or name...'} />
    <SearchButton onClick={() => console.log('pressed search')}>
      <SearchIcon />
    </SearchButton>
  </Container>
);

export default Search;
