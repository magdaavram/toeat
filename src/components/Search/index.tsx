import React from 'react';
import styled from 'styled-components';
import Button from 'components/Button';
import Input from 'components/Input';

interface IProps {
  handleClick: () => void;
}

const Container = styled.div`
  display: flex;
  width: 70%;
  max-width: 1000px;
  height: 54px;
`;

const SearchInput = styled(Input)`
  height: 100%;
  border-width: 3px;
  border-radius: 12px;
  padding: 0 54px 0 18px;
`;

const SearchButton = styled(Button)`
  width: 30px;
  height: 30px;
  margin: auto 0;
  margin-left: -45px;
  opacity: 0.6;
`;

const Search = (props: IProps) => (
  <Container>
    <SearchInput placeholder={"Find recipes by ingredients or name..."}/>

    <SearchButton
      onClick={props.handleClick}
      hasIcon={true}
      icon={"search"}
      iconWidth={"30px"}
      iconHeight={"30px"}
    />
  </Container>
);

export default Search;
