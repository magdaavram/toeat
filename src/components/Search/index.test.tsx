import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import Search from './index';

it('should render component correctly', () => {
  const onClick = jest.fn();
  const tree = create(<Search handleClick={onClick} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should call the onClick function', () => {
  const onClick = jest.fn();
  const wrapper = mount(<Search handleClick={onClick} />);

  wrapper.find('button').simulate('click');
  expect(onClick).toHaveBeenCalledTimes(1);
});
