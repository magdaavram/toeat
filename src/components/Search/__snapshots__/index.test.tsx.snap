// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
<div
  className="sc-AxiKw gZTYQF"
>
  <input
    className="sc-AxirZ sc-AxhCb bHWOmi"
    placeholder="Find recipes by ingredients or name..."
  />
  <button
    className="sc-AxjAm sc-AxhUy iYVaFW"
    onClick={[MockFunction]}
    type="button"
  >
    <svg
      height="30"
      viewBox="0 0 30 30"
      width="30"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <image
          height="30"
          href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB4AAAAeCAMAAAAM7l6QAAAAAXNSR0IB2cksfwAAARRQTFRFAAAASk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pSk5pVwjv/QAAAFx0Uk5TADWDv+P19ufLh0YFAyuM2vr/++GfOQJf1/7qUFvYWDIvTHX590sBMLQ8IKruPhG34pmlTdMWLOvoKRnzMxg0JuUbi6gE0E61/Mx5VBKakwv9DT+QsQeSdwgPsNTBV9X7AAABhUlEQVR4nHWS61aCQBSF8RokBhutxJq0TNRMK6+V3bQ0ukMqqb3/ezQ1BjhrsX/tMx+Lc+bsEQSmUDgSjcXXRGk9IfCSkxuKij9pqfTmFoe3Mzo8ZXdWf7Cr0UMi7eXy+wdiATgM+7i8TSkpGqwySmUdmiS7OJkBxIrbzziqAsc1F2/oIBV/rzoBTv6LkAIUV2ctnkIPLX1YBTFWL2KcAY1l9wgg8Xtoqmi1mY0CezzudHF+wWwMyPH4kqB3xWwcyPP4OoPeDbNrQJPHt3foJpkVgQMe1+JQ+sxKGkT+YgPg/oH59RQKpVVeGkIfLX0iDTwe+elW2cSTm8lmFnq17qPPOswXr9w5pIHu+wMFefU+T4Rp4NG3Zufy+rY2GJr0bajvHz4uHf8edUnmji4JJlFhpW2Py58n7mPTn15e3y1YY984NPZG67zX6yr3IzrzR9qCNpn6udy+uLpJ9tk27LEGy7GFINUnFtSvWSCfOnQ+Zx7I7S8L5iIQCzPHhBqMhflC/f4BU6Uw/3BpkC8AAAAASUVORK5CYII="
          id="search-icon"
          width="30"
        />
      </defs>
      <use
        href="#search-icon"
        x="0"
        y="0"
      />
    </svg>
  </button>
</div>
`;
